src/rows.ts

```diff
@@ -39,62 +38,62 @@ export class Rows {
   }
 
   /**
-   * Returns a readable stream of {@link Row}s for responses
+   * Returns an async iterator of {@link Row}s for responses
    * in {@link StreamableDataFormat} format.
    *
-   * The method will throw if called on a response in non-streamable format,
-   * and if the underlying stream was already consumed
-   * by calling the other methods
+   * If selected format is non-streamable,
+   * or the underlying stream was already consumed,
+   * it will throw when accessing the next element
    */
-  stream(): Stream.Readable {
-    // If the underlying stream has already ended by calling `text` or `json`,
-    // Stream.pipeline will create a new empty stream
-    // but without "readableEnded" flag set to true
+  async *stream(): AsyncGenerator<Row, void> {
     if (this._stream.readableEnded) {
       throw Error(streamAlreadyConsumedMessage)
     }
-
     validateStreamFormat(this.format)
+    const textDecoder = new TextDecoder()
+    let decodedChunk = ''
+    for await (const chunk of this._stream) {
+      decodedChunk += textDecoder.decode(chunk, { stream: true })
+      let idx = 0
+      while (true) {
+        idx = decodedChunk.indexOf('\n')
+        if (idx !== -1) {
+          const line = decodedChunk.slice(0, idx)
+          decodedChunk = decodedChunk.slice(idx + 1)
+          yield {
+            /**
+             * Returns a string representation of a row.
+             */
```

Comment:
These comments belong to `Row` interface declared below. https://github.com/ClickHouse/clickhouse-js/pull/108/files#diff-b13826a37e4f93783b49eaca9c60dc1d124ee9d6b331be22244f41cc7bb09d39R94-R96
IDE doesn't hint at the method signature.